panopticum/management/commands/importer.py

```diff
@@ -0,0 +1,1085 @@
+                                         description=description, order=order)
+
+    @staticmethod
+    def seed_comp_categories():
+        # fn_name = 'Utility::seed_comp_categories() -'
+        logging.info('starts')
+        for (name, values) in Utility.comp_categories.items():
+            dbobj = Utility.get_comp_category(name, description=values['description'], order=values['order'])
+            if dbobj is None:
+                pass  # TODO: Exception
+
+    @staticmethod
+    def get_comp_subcategory(name, category_id, description=''):  # Creates record in DB if doesn't exists.
+        # fn_name = 'Utility::get_comp_subcategory() -'
+        sub_category_obj = None
+        created = False
+        try:
+            sub_category_obj, created = ComponentSubcategoryModel.objects\
+                .get_or_create(name=name,
+                               category_id=category_id,
+                               defaults={'description': description})
+        except Exception as e:
+            logging.error('Ignoring exception %s', e)
+
+        if created is True:  # We can now safely create the missing object
+            # sub_category_obj = ComponentSubcategoryModel(name=name, description=description,
+            #                                              category_id=category_id)
+            # sub_category_obj.save()
+            logging.info("Created in model [%s] record - [%s]",
+                          ComponentSubcategoryModel.__name__,
+                         serializers.serialize('json', [sub_category_obj, ]))
+        return sub_category_obj
+
+    @staticmethod
+    def seed_comp_subcategories():
+        # fn_name = 'Utility::seed_comp_subcategories() -'
+        logging.info('starts')
+        Utility.seed_comp_categories()  # IMP: Required step for below code to work
+        for comp_subcategory in Utility.comp_subcategories:
+            name = comp_subcategory['name']
+            category_name = comp_subcategory['category']
+            category_obj = Utility.get_comp_category(category_name)
+            # category_obj = obj[0]
+            if category_obj is None:
+                logging.info('Skipping Value! [%s] couldn\'t get from ComponentCategoryModel DB table.'
+                             , category_name)
+                continue  # TODO: raise exception and handle in caller code
+            Utility.get_comp_subcategory(name, category_obj.id, comp_subcategory['description'])
+
+    @staticmethod
+    def get_comp_privacy(name, order=1):  # Creates record in DB if doesn't exists.
+        return Utility.alloc_obj_by_name(ComponentDataPrivacyClassModel, name, order=order)
+
+    @staticmethod
+    def seed_comp_privacies():
+        # fn_name = 'Utility::seed_comp_privacies() -'
+        logging.info('starts')
+        for (name, values) in Utility.comp_dataprivacies.items():
+            dbobj = Utility.get_comp_privacy(name, values['order'])

```

Comment:
You can probably add "raise_exception" parameter to get_comp_privacy() and optimize all the cases with:

obj = Utility.get_comp_privacy(..)
if obj is None:
     pass   # TODO: Exception

Reply:
Got it, will add and let utility functions raise those exceptions.